### PR TITLE
add playbooks/roles for aws efs storage and manage mmi monitoring configuration

### PR DESCRIPTION
**Pull request by bzzwh**
add playbooks/roles for aws efs storage and manage mmi monitoring configuration

### CODE REVIEW

File: ibm/mas_devops/roles/ocp_setup_mas_deps/defaults/main.yml [hunk 1]
```diff
@@ -1,25 +0,0 @@
----
```
Comment (durera) [inline ibm/mas_devops/roles/ocp_setup_mas_deps/defaults/main.yml:1]: Why was this file deleted?

File: ibm/mas_devops/roles/ocp_setup_aws_efs/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ocp_setup_aws_efs
+==================
+
+This role provides support to configure AWS EFS based storageclass into Openshift cluster.
```
Comment (durera) [inline ibm/mas_devops/roles/ocp_setup_aws_efs/README.md:4]: Can you elaborate on what this means?  What are potential uses for the AWS EFS storageclass within MAS.  Also please document variables, e.g. see examples and a sample playbook for the role ... see https://github.com/ibm-mas/ansible-devops/blob/master/ibm/mas_devops/roles/mongodb/README.md for example of the expected format of the README

File: ibm/mas_devops/roles/manage_mmi_monitoring_configure/templates/podmonitor.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  podMetricsEndpoints:
+  - interval: 30s
+    targetPort: 9081
+    scheme: http
+  selector:
+    mas.ibm.com/appType: serverBundle
+    mas.ibm.com/appTypeName: {{ manage_app_type_name}}
+    mas.ibm.com/applicationId: manage
+    mas.ibm.com/instanceId: {{ mas_instance_id }}
+    mas.ibm.com/workspaceId: {{ mas_workspace_id }}
```
Comment (durera) [inline ibm/mas_devops/roles/manage_mmi_monitoring_configure/templates/podmonitor.yml:18]: This will fail linting, there need to be a newline at end of yaml files

File: ibm/mas_devops/playbooks/ocp/configure-efs.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+- hosts: localhost
+  any_errors_fatal: true
+  vars:
+    file_system_id: "{{ lookup('env', 'FILE_SYSTEM_ID') }}"
+    aws_region: "{{ lookup('env', 'AWS_REGION') | default('us-east-1', true) }}"
+  roles:
+    - ibm.mas_devops.ocp_setup_aws_efs
```
Comment (durera) [inline ibm/mas_devops/playbooks/ocp/configure-efs.yml:8]: no newline at EOF will fail linting

File: ibm/mas_devops/playbooks/mas/configure-manage-mmi-monitoring.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    prometheus_storage_class: "{{ lookup('env', 'PROMETHEUS_STORAGE_CLASS') }}"
+    prometheus_storage_size: "{{ lookup('env', 'PROMETHEUS_STORAGE_SIZE') }}"
+    prometheus_retention_period: "{{ lookup('env', 'PROMETHEUS_RETENTION_PERIOD') | default('90d',true)}}"
+
+    # Alertmgr
+    prometheus_alertmgr_storage_class: "{{ lookup('env', 'PROMETHEUS_ALERTMGR_STORAGE_CLASS') }}"
+    prometheus_alertmgr_storage_size: "{{ lookup('env', 'PROMETHEUS_ALERTMGR_STORAGE_SIZE') }}"
+
+  roles:
+    - ibm.mas_devops.manage_mmi_monitoring_configure
```
Comment (durera) [inline ibm/mas_devops/playbooks/mas/configure-manage-mmi-monitoring.yml:24]: no newline at EOF will fail linting

File: ibm/mas_devops/roles/manage_mmi_monitoring_configure/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+Role Name
```
Comment (durera) [inline ibm/mas_devops/roles/manage_mmi_monitoring_configure/README.md:1]: This readme must be completed before this is merged.  Provide details about what this role is doing and why it would be used/what a user can do after they do this, e.g. explain basically what MMI monitoring is (what does MMI stand for), what it gives the user that they didn't have before they run this role, what they need to do to see the results of this monitoring (e.g. is there a dashboard somewhere, how do they find out the URL of the dashboard) etc

**Review by durera — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/ocp_setup_mas_deps/defaults/main.yml (hunk 1 above)
Comment (bzzwh) [reply ibm/mas_devops/roles/ocp_setup_mas_deps/defaults/main.yml:1]: the file was deleted by mistake. add it back.

File: ibm/mas_devops/playbooks/mas/configure-manage-mmi-monitoring.yml (hunk 5 above)
Comment (bzzwh) [reply ibm/mas_devops/playbooks/mas/configure-manage-mmi-monitoring.yml:24]: fixed.

File: ibm/mas_devops/roles/manage_mmi_monitoring_configure/README.md (hunk 6 above)
Comment (bzzwh) [reply ibm/mas_devops/roles/manage_mmi_monitoring_configure/README.md]: README is updated.

**Comment by whitfiea** [comment]
@bzzwh I had some comments on this PR:

The ocp_setup_aws_efs role seems to be just a call to a bash script that then creates various resources. These should all be created using ansible rather than via a script unless there is something in the script which can not be done via ansible.

The mmi_monitoring seems to be configuring prometheus userworkload storage which is likely to already be setup for other apps, and so this manage specific one will overwrite that. Also the PodMonitor should really be provided by the MAS manage operator like other apps in MAS provide ServiceMonitors.

**Comment by durera** [comment]
Closing PR as there has been no response to requests for updates and this branch is now significantly behind master
